### PR TITLE
ACEReader fixes 

### PR DESCRIPTION
**Pull request by bhargav**
Fixes towards #160. Simplifies ACE Reader parsing, cleanup unused code and remove some metadata content from **TOKENS** view. @mssammon 

- Replaces `IllinoisTokenizer` with `StatefulTokenizer`
- All Entities and Relations are populated in a single view now. Removes all older views [**This breaks backwords compatibility for ACEReader**]
- Entity Constituents represent EXTENT of the mention. Character Offsets for HEAD is stores as an attribute.
- Relations are represented as `Relation`s in the Entity View directed from "Arg-1" to "Arg-2" role implicitly.
- Entities and Relations have the coarse type as label and fine type is added to attributes.

### CODE REVIEW

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java [hunk 1]
```diff
@@ -131,10 +118,7 @@ public static ViewTypes getViewType(String viewName) {
                 return ViewTypes.TOKEN_LABEL_VIEW;
             case SENTENCE:
             case PARAGRAPH:
-            case NER_ACE_COARSE_HEAD:
-            case NER_ACE_FINE_HEAD:
-            case NER_ACE_COARSE_EXTENT:
-            case NER_ACE_FINE_EXTENT:
+            case NER_ACE:
```
Comment (mssammon) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java]: I think for clarity we should call this "MENTION_ACE" as it has non-Named Entity mentions (correct me if I'm wrong)

**Comment by mssammon** [comment]
@bhargav please see line note about mention view name. 
doc/ACEReader.md needs to be updated to reflect these improvements.
Also: does it make sense to store the metadata fields you now ignore as attribute values in TextAnnotation? (If yes but not urgent, this could be a TODO issue)

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java (hunk 1 above)
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/ViewNames.java]: Updated the name. It is not strictly NER but also has mentions.

**Comment by bhargav** [comment]
@mssammon Updated the documentation as well.
 
Regarding the ignored attributes, I'll add that as attributes in TextAnnotation. Will send it as a separate PR as it is not urgent right now.

**Comment by mssammon** [comment]
Great -- will merge on green.

**Pull request merged**